var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -681,14 +683,25 @@ def define_enable(suffix, value=None):
         # ################# System-specific ######################
 
         # Fortran lib (assumes clang is built with gfortran!)
-        if ('+fortran +mpi' in spec
+        if ('+fortran' in spec
                 and spec.compiler.name in ['gcc', 'clang', 'apple-clang']):
-            mpifc = Executable(spec['mpi'].mpifc)
-            libgfortran = mpifc('--print-file-name', 'libgfortran.a', output=str)
-            options.append(define(
-                'Trilinos_EXTRA_LINK_FLAGS',
-                '-L%s/ -lgfortran' % (libgfortran),
-            ))
+            fc = Executable(spec['mpi'].mpifc) if (
+                '+mpi' in spec) else Executable(spack_fc)
+            sharedlibext = 'dylib' if sys.platform == 'darwin' else 'so'
```

Comment:
Spack has to do this elsewhere, so maybe just add
```python
from spack.build_environment import dso_suffix
```
to the imports and use `dso_suffix` here?